### PR TITLE
feat: Speed up xpath lookup by only including the requested attributes

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
Querying view attributes is expensive and takes time. Although xpath lookup only requires some attributes to be there and does not need them all

### CODE REVIEW

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/model/SourceDocument.kt [hunk 1]
```diff
@@ -130,34 +130,75 @@ class SourceDocument @JvmOverloads constructor(
...
-        setAttribute(ViewAttributesEnum.LONG_CLICKABLE, viewElement.isLongClickable)
-        setAttribute(ViewAttributesEnum.PASSWORD, viewElement.isPassword)
-        setAttribute(ViewAttributesEnum.SELECTED, viewElement.isSelected)
-        setAttribute(ViewAttributesEnum.VISIBLE, viewElement.isVisible)
-        setAttribute(ViewAttributesEnum.BOUNDS, viewElement.bounds.toShortString())
-        val viewText = viewElement.text
-        viewText?.let {
-            setAttribute(ViewAttributesEnum.TEXT, it.rawText)
-            setAttribute(ViewAttributesEnum.HINT, it.isHint)
+        if (null == includedAttributes || includedAttributes.contains(ViewAttributesEnum.INDEX)) {
```
Comment (KazuCocoa) [inline espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/model/SourceDocument.kt:138]: How about define a variable for `null == includedAttributes`?
```kotlin
val shouldSetAllAttributes = null == includedAttributes
if (shouldSetAllAttributes || includedAttributes.contains(ViewAttributesEnum.INDEX)) {
  // ...
}
```

**Review by KazuCocoa — APPROVED** [review]
I left a preference comment, otherwise lgtm

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/model/SourceDocument.kt (hunk 1 above)
Comment (mykola-mokhnach) [reply espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/model/SourceDocument.kt]: I've refactored it in more functional style

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/model/SourceDocument.kt [hunk 2]
```diff
@@ -117,48 +116,67 @@ class SourceDocument @JvmOverloads constructor(
...
-            setAttribute(ViewAttributesEnum.TEXT, it.rawText)
-            setAttribute(ViewAttributesEnum.HINT, it.isHint)
-        }
-        setAttribute(ViewAttributesEnum.RESOURCE_ID, viewElement.resourceId)
-        setAttribute(ViewAttributesEnum.VIEW_TAG, viewElement.viewTag)
-        if (view is AdapterView<*>) {
-            recordAdapterViewInfo(view)
+        var isTextOrHintRecorded = false
+        var isAdapterInfoRecorded = false
+        linkedMapOf(
```
Comment (KazuCocoa) [inline espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/model/SourceDocument.kt:140]: 👍 👍

**Review by KazuCocoa — APPROVED** [review]

**Review by KazuCocoa — APPROVED** [review]

**Pull request merged**